homeassistant/components/life360/__init__.py

```diff
@@ -28,184 +22,174 @@
-            continue
-        if (
-            not password
-            and entry.source == config_entries.SOURCE_IMPORT
-            or password
-            and password != entry.data[CONF_PASSWORD]
-        ):
-            hass.async_create_task(hass.config_entries.async_remove(entry.entry_id))
-
-    # Create config entries for accounts listed in configuration.
-    for account in conf[CONF_ACCOUNTS]:
-        if account[CONF_USERNAME] not in already_configured:
-            hass.async_create_task(
-                hass.config_entries.flow.async_init(
-                    DOMAIN,
-                    context={"source": config_entries.SOURCE_IMPORT},
-                    data=account,
-                )
-            )
+    if entry.version == 1:
+        entry.version = 2
+        entry.source = SOURCE_USER
+        unique_id = entry.data[CONF_USERNAME].lower()
+        hass.config_entries.async_update_entry(
+            entry,
+            unique_id=unique_id,
+            title=unique_id,
+            options=hass.data[DOMAIN]["cfg_options"],
+        )
+
+    LOGGER.info("Config entry migration to version %s successful", entry.version)
     return True
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up config entry."""
-    hass.data[DOMAIN]["apis"][entry.data[CONF_USERNAME]] = get_api(
-        entry.data[CONF_AUTHORIZATION]
+    account = hass.data[DOMAIN]["accounts"].setdefault(
+        cast(str, entry.unique_id), AccountData()
     )
+
+    if not (api := account.get("api")):
+        api = get_life360_api(authorization=entry.data[CONF_AUTHORIZATION])
+        account["api"] = api
+
+    async def async_update_data() -> dict[str, dict[str, Any]]:
+        """Update Life360 data."""
+        return await get_life360_data(hass, api)
+
+    if not (coordinator := account.get("coordinator")):
+        coordinator = account["coordinator"] = DataUpdateCoordinator(
+            hass,
+            LOGGER,
+            name=f"{DOMAIN} ({entry.unique_id})",
+            update_interval=_update_interval(entry),
+            update_method=async_update_data,
+        )
+
+    await coordinator.async_config_entry_first_refresh()
+
+    # Set up components for our platforms.
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    # Add event listener for option flow changes
+    entry.async_on_unload(entry.add_update_listener(async_update_options))
+
     return True
 
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload config entry."""
+    # Unload components for our platforms.
+    # But first stop checking for new members on update.
+    if unsub := hass.data[DOMAIN]["accounts"][entry.unique_id].pop("unsub", None):
+        unsub()
+    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+async def async_remove_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
```

Comment:
We should remove the config entry data from hass.data when unloading the config entry.

Reply:
But there's still needed data in there for when the entry is reloaded. I remove the data associated with this entry from hass.data when the config entry is removed. (See line 181.)

Reply:
We should not reuse the coordinator when the config entry is reloaded.